htsget_server/beacon_operations.py

```diff
@@ -0,0 +1,600 @@
+
+
+def compile_beacon_resultset(variants_by_obj, reference_genome="hg38"):
+    """
+    Each beacon result describes a variation at a specific position:
+    resultset = [
+        {
+          "caseLevelData": [...],
+          "identifiers": {
+            "genomicHGVSId": "NC_000021.8:g.5031153="
+          },
+          "variantInternalId": "NC_000021.8:g.5031153=",
+          "variation": {
+            "location": {
+              "interval": {
+                "end": {
+                  "type": "Number",
+                  "value": 5031153
+                },
+                "start": {
+                  "type": "Number",
+                  "value": 5031152
+                },
+                "type": "SequenceInterval"
+              },
+              "sequence_id": "refseq:NC_000021.8",
+              "type": "SequenceLocation"
+            },
+            "state": {
+              "sequence": "G",
+              "type": "LiteralSequenceExpression"
+            },
+            "type": "Allele"
+          }
+        }
+      ]
+    """
+    resultset = {}
+    for drs_obj in variants_by_obj.keys():
+        # check to see if this drs_object is authorized:
+        x, status_code = drs_operations.get_object(drs_obj)
+        is_authed = (status_code == 200)
+
+        for variant in variants_by_obj[drs_obj]['variants']:
+            # parse the variants beacon-style
+            variant['variations'] = compile_variations_from_record(ref=variant.pop('ref'), alt=variant.pop('alt'), chrom=variant.pop('chrom'), pos=variant.pop('pos'), reference_genome=reference_genome)
+            assign_info_to_variations(variant)
+
+            # the variations in each variant need to be copied out first:
+            resultset[drs_obj] = []
+            for var in variant['variations']:
+                resultset[drs_obj].append(var['hgvsid'])
+                if var['hgvsid'] not in resultset:
+                    resultset[var['hgvsid']] = {
+                        'variation': {
+                            "location": var.pop('location'),
+                            "state": var.pop('state'),
+                            "type": var.pop('type')
+                        },
+                        "identifiers": {
+                            "genomicHGVSId": var['hgvsid']
+                        }
+                    }
+                # move allele-specific info to the variant, like CSQ annotations
+                if 'info' in var:
+                    if 'CSQ' in var['info']:
+                        if 'molecularAttributes' not in resultset[var['hgvsid']]:
+                            compile_molecular_attributes_from_csq(resultset[var['hgvsid']], var['info'].pop('CSQ'))
+
+            # now process the samples into the variations:
+            if 'samples' in variant and len(variant['samples']) > 0:
+                for k in variant['samples'].keys():
+                    sample = variant['samples'][k]
+                    cld = {
```

Comment:
```suggestion
                    # Begin creating a Case Level Data object
                    cld = {
```
(For readability; wasn't sure what a cld was until I saw the spec above)